### PR TITLE
i3-compat: add GET_BINDING_STATE IPC command

### LINKED ISSUE #5447: [i3-compat] Add new IPC command GET_BINDING_STATE
i3 is adding a `GET_BINDING_STATE` command to get the currently active binding mode, which should be supported.

PRs: i3/i3#4122 for IPC, i3/i3#4128 for i3-msg.

### PR DESCRIPTION
**Pull request by ammgws**
Introduced to i3 here https://github.com/i3/i3/pull/4122.

Resolves #5447

### CODE REVIEW

File: sway/ipc-json.c [hunk 1]
```diff
@@ -1261,3 +1261,9 @@ json_object *ipc_json_describe_bar_config(struct bar_config *bar) {
 #endif
 	return json;
 }
+
+json_object *ipc_json_get_binding_mode(void) {
+	json_object *current_mode = json_object_new_object();
+	json_object_object_add(current_mode, "name", json_object_new_string(config->current_mode->name));
```
Comment (Xyene) [inline sway/ipc-json.c:1267]: Nit: try to keep lines to 80 characters where possible.

File: completions/fish/swaymsg.fish [hunk 2]
```diff
@@ -18,6 +18,7 @@ complete -c swaymsg -s t -l type -fra 'get_marks' --description "Get a JSON-enco
 complete -c swaymsg -s t -l type -fra 'get_bar_config' --description "Get a JSON-encoded configuration for swaybar."
 complete -c swaymsg -s t -l type -fra 'get_version' --description "Get JSON-encoded version information for the running instance of sway."
 complete -c swaymsg -s t -l type -fra 'get_binding_modes' --description "Gets a JSON-encoded list of currently configured binding modes."
+complete -c swaymsg -s t -l type -fra 'get_binding_state' --description "Get JSON-encoded info about the current binding state"
```
Comment (Xyene) [inline completions/fish/swaymsg.fish:21]:
```suggestion
complete -c swaymsg -s t -l type -fra 'get_binding_state' --description "Get JSON-encoded info about the current binding state."
```

Comment (ammgws) [reply completions/fish/swaymsg.fish]: I changed it for now but fish itself removes trailing periods and has less verbose descriptions for completions it ships so I may address that in another PR.

File: sway/ipc-json.c (hunk 1 above)
Comment (ammgws) [reply sway/ipc-json.c]: Rebased with style changes.

File: swaymsg/main.c [hunk 3]
```diff
@@ -440,6 +440,8 @@ int main(int argc, char **argv) {
 		type = IPC_GET_VERSION;
 	} else if (strcasecmp(cmdtype, "get_binding_modes") == 0) {
 		type = IPC_GET_BINDING_MODES;
+	} else if (strcasecmp(cmdtype, "get_binding_state") == 0) {
```
Comment (RedSoxFan) [inline swaymsg/main.c:443]: Can you please also add this to `swaymsg.1.scd`?

File: completions/fish/swaymsg.fish (hunk 2 above)
Comment (Xyene) [reply completions/fish/swaymsg.fish]: Sounds good :+1:

File: completions/fish/swaymsg.fish [hunk 4]
```diff
@@ -18,6 +18,7 @@ complete -c swaymsg -s t -l type -fra 'get_marks' --description "Get a JSON-enco
 complete -c swaymsg -s t -l type -fra 'get_bar_config' --description "Get a JSON-encoded configuration for swaybar."
 complete -c swaymsg -s t -l type -fra 'get_version' --description "Get JSON-encoded version information for the running instance of sway."
 complete -c swaymsg -s t -l type -fra 'get_binding_modes' --description "Gets a JSON-encoded list of currently configured binding modes."
+complete -c swaymsg -s t -l type -fra 'get_binding_state' --description "Get JSON-encoded info about the current binding state."
```
Comment (Xyene) [inline completions/fish/swaymsg.fish:21]: I suppose on this note the bash and zsh completions should probably be updated, too.

File: swaymsg/main.c (hunk 3 above)
Comment (ammgws) [reply swaymsg/main.c:443]: Updated swaymsg.1.scd and rebased

File: completions/fish/swaymsg.fish (hunk 4 above)
Comment (ammgws) [reply completions/fish/swaymsg.fish:21]: Seemed simple enough so I updated those too and rebased.

**Comment by RedSoxFan** [comment]
Thanks!

**Pull request merged**